### PR TITLE
Feat/msdk 594 abdicate

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/procedures/__tests__/waivePermissions.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    entityMockUtils.cleanup();
+    procedureMockUtils.cleanup();
+    dsMockUtils.cleanup();
+  });
+
+  test('should throw an error if the Identity is not part of the Agent Group', async () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/waivePermissions.ts:61]: is not an Agent for the Security Token

File: src/api/procedures/__tests__/waivePermissions.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }),
+      });
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error.message).toBe("The Identity is not part of the token's Agent Group");
+  });
+
+  test('should add a abdicate transaction to the queue', async () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/waivePermissions.ts:93]:
```suggestion
  test('should add an abdicate transaction to the queue', async () => {
```

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/waivePermissions.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * @hidden
+ */
+export interface WaivePermissionsParams {
+  token: string | SecurityToken;
+}
+
+export type Params = WaivePermissionsParams & {
+  identity: Identity;
+};
```
[the remark this answers is not in the document]
Comment (shuffledex) [reply src/api/procedures/waivePermissions.ts:16]: my mistake

File: src/api/procedures/__tests__/waivePermissions.ts [hunk 4]
```diff
@@ -58,7 +54,7 @@ describe('waivePermissions procedure', () => {
     dsMockUtils.cleanup();
   });
 
-  test('should throw an error if the Identity is not part of the Agent Group', async () => {
+  test('should throw an error if the Identity is not is not an Agent for the Security Token', async () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/waivePermissions.ts:57]:
```suggestion
  test('should throw an error if the Identity is not an Agent for the Security Token', async () => {
```

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by monitz87 — APPROVED** [review]

**Review by monitz87 — CHANGES_REQUESTED** [review]
We still have to figure out the permissions issue

**Review by monitz87 — APPROVED** [review]

**Pull request merged**